go/cmd/dolt/commands/sql.go

```diff
@@ -580,64 +604,126 @@ func processBatchQuery(ctx context.Context, query string, se *sqlEngine) error {
 		return fmt.Errorf("Error parsing SQL: %v.", err.Error())
 	}
 
-	switch sqlStatement.(type) {
-	case *sqlparser.Insert:
+	if canProcessAsBatchInsert(sqlStatement) {
```

Comment:
Seems preferable to put all this code into a separate function that will have it's defered blocks called on exit rather than leaving them dangling
.

Reply:
I extracted a couple functions here. But I did it for me!